### PR TITLE
Support for new feed-consumer plugin - Apache Pulsar

### PR DESCRIPTION
**Pull request by wazir-ahmed**
Signed-off-by: Wazir Ahmed <wazir@accuknox.com>

### CODE REVIEW

File: deployments/k8s/deployment.yaml [hunk 1]
```diff
@@ -75,19 +75,19 @@ data:
...
-          enabled: false
-        events:
-          buffer: 50
+      driver: "pulsar"
+      servers:
+        - "pulsar-proxy.accuknox-dev-pulsar.svc.cluster.local:6650"
+      topic: 
+        cilium: "persistent://accuknox/datapipeline/ciliumalertsflowv1"
+        kubearmor: "persistent://accuknox/datapipeline/kubearmoralertsflowv1"
+      encryption:
```
Comment (wazir-ahmed) [inline deployments/k8s/deployment.yaml:84]: Format has been changed slightly.

File: src/conf/local-file.yaml [hunk 2]
```diff
@@ -54,3 +54,40 @@ cilium-hubble:
...
+  encryption:
+    enable: false
+    ca-cert: /kafka-ssl/ca.pem 
+  auth:
+    enable: false
+    cert: /kafka-ssl/user.cert.pem
+    key: /kafka-ssl/user.key.pem
+    # keystore:
+    #   path: /kafka-ssl/user.p12
+    #   password: DHmHwNYy22At
```
Comment (wazir-ahmed) [inline src/conf/local-file.yaml:79]: Used in case of `kakfa`.

File: src/conf/local-file.yaml [hunk 3]
```diff
@@ -54,3 +54,40 @@ cilium-hubble:
...
+    kubearmor: "persistent://accuknox/datapipeline/kubearmoralertsflowv1"
+    # cilium: "cilium-alerts"
+    # kubearmor: "kubearmor-alerts"
+  encryption:
+    enable: false
+    ca-cert: /kafka-ssl/ca.pem 
+  auth:
+    enable: false
+    cert: /kafka-ssl/user.cert.pem
+    key: /kafka-ssl/user.key.pem
```
Comment (wazir-ahmed) [inline src/conf/local-file.yaml:76]: Used in case of `pulsar`.

File: src/conf/local-file.yaml [hunk 4]
```diff
@@ -54,3 +54,40 @@ cilium-hubble:
...
+  # number-of-consumers: 1
+  # consumer-group: knoxautopolicy
+  # event-buffer-size: 50
+  # kafka:
+  #   server-address-family: v4
+  #   session-timeout: 6000 # in millisecond
+  # pulsar:
+  #   connection-timeout: 10 # in second
+  #   operation-timeout: 30 # in second
+  # -----------------------------------------------
```
Comment (wazir-ahmed) [inline src/conf/local-file.yaml:96]: I have made these values a kind of hidden configs.

File: src/networkpolicy/networkPolicy.go [hunk 5]
```diff
@@ -2147,8 +2147,10 @@ func StartNetworkLogRcvr() {
 	for {
 		if cfg.GetCfgNetworkLogFrom() == "hubble" {
 			plugin.StartHubbleRelay(NetworkStopChan /* &NetworkWaitG, */, cfg.GetCfgCiliumHubble())
-		} else if cfg.GetCfgNetworkLogFrom() == "kafka" {
-			feedconsumer.StartConsumer()
+		} else if cfg.GetCfgNetworkLogFrom() == "feed-consumer" {
+			fc.ConsumerMutex.Lock()
+			fc.StartConsumer()
+			fc.ConsumerMutex.Unlock()
```
Comment (wazir-ahmed) [inline src/networkpolicy/networkPolicy.go:2172]: This lock here is necessary since the `fc.StartConsumer()` updates few global variables.

**Review by seswarrajan — APPROVED** [review]
LGTM.

**Comment by wazir-ahmed** [comment]
@nyrahul This PR is tested with both Kafka and Pulsar in the saas-dev environment. `knoxautopolicy` is able to fetch the feeds from both the sources and process it. We can go ahead and merge this. Once the datapipeline team fixes the TCP issue in the `feeder-service`, things will work just fine.

**Review by nyrahul — APPROVED** [review]
LGTM

**Pull request merged**